src/core/libraries/fiber/fiber.cpp

```diff
@@ -6,262 +6,449 @@
 #include "common/logging/log.h"
 #include "core/libraries/fiber/fiber_error.h"
 #include "core/libraries/libs.h"
-#include "core/tls.h"
-
-#ifdef _WIN64
-#include <windows.h>
-#endif
 
 namespace Libraries::Fiber {
 
-static constexpr u64 kFiberSignature = 0x054ad954;
+static constexpr u32 kFiberSignature0 = 0xdef1649c;
+static constexpr u32 kFiberSignature1 = 0xb37592a0;
+static constexpr u32 kFiberOptSignature = 0xbb40e64d;
+static constexpr u64 kFiberStackSignature = 0x7149f2ca7149f2ca;
+static constexpr u64 kFiberStackSizeCheck = 0xdeadbeefdeadbeef;
+
+thread_local OrbisFiberContext* gFiberCtx = nullptr;
+static std::atomic<u32> contextSizeCheckEnabled = false;
 
-thread_local SceFiber* gCurrentFiber = nullptr;
-thread_local void* gFiberThread = nullptr;
+extern "C" s32 PS4_SYSV_ABI _sceFiberSetJmp(OrbisFiberContext* ctx);
+extern "C" s32 PS4_SYSV_ABI _sceFiberLongJmp(OrbisFiberContext* ctx);
+extern "C" void PS4_SYSV_ABI _sceFiberSwitchEntry(OrbisFiberData* data, bool set_fpu);

```

Comment:
```suggestion
extern "C" s32 PS4_SYSV_ABI _sceFiberSetJmp(OrbisFiberContext* ctx) asm("_sceFiberSetJmp");
extern "C" s32 PS4_SYSV_ABI _sceFiberLongJmp(OrbisFiberContext* ctx) asm("_sceFiberLongJmp");
extern "C" void PS4_SYSV_ABI _sceFiberSwitchEntry(OrbisFiberData* data,
                                                  bool set_fpu) asm("_sceFiberSwitchEntry");
extern "C" void PS4_SYSV_ABI _sceFiberForceQuit(u64 ret) asm("_sceFiberForceQuit");
```

macOS CI is failing because it follows the old convention of prepending non-ASM symbols with an underscore. Fixable by adding specifiers for the assembly symbol names.

Reply:
Thank you, I was wondering why it wasn't compiling the assembly.

Reply:
It's needed for `_sceFiberForceQuit` as well for the assembly to see it, which is why I included it here. As a separate line from the existing declaration since you can't do the `asm` specifier with the body.

Reply:
How should it look then?

Reply:
Exactly as in the suggested change, just replace the 3 lines from before with these 4 lines and no other changes.

Reply:
Ah, so the declaration with asm above and the implementation below like this?
```cpp
extern "C" void PS4_SYSV_ABI _sceFiberForceQuit(u64 ret) asm("_sceFiberForceQuit");

extern "C" void PS4_SYSV_ABI _sceFiberForceQuit(u64 ret) {
    gFiberCtx->returnVal = ret;
    _sceFiberLongJmp(gFiberCtx);
}
```

Reply:
Yes